tsfc/driver.py

```diff
@@ -284,8 +290,19 @@ def compile_expression_dual_evaluation(expression, to_element, coordinates, *,
     """
     import coffee.base as ast
     import loopy as lp
-    if any(len(dual.deriv_dict) != 0 for dual in to_element.dual_basis()):
-        raise NotImplementedError("Can only interpolate onto dual basis functionals without derivative evaluation, sorry!")
+
+    # TODO: Temporary fix to maintain old code, fix mapping name
+    if isinstance(element, FiniteElementBase):
+        # FInAT mapping outputs string, FIAT mapping outputs tuple
+        temp_mapping = (element.mapping,)
+        if isinstance(element, FiatElement):
+            to_element = element._element
```

Comment:
Note that you can end up not actually setting `to_element` here. I have tried to run this with a FInAT `QuadratureElement` and can't get past line 417.

Reply:
Where are you calling `compile_expression_dual_evaluation` from? If `interpolate` is being used it should be passed a FIAT element instead of a FInAT element for `QuadratureElement`, though I agree there should be better checking (and more consistent class types) here.

Reply:
From within `interpolate` - I will add the required `dual_evaluation` and `dual_basis` methods

Reply:
Only `dual_basis` should be required unless it requires some wonky delta cancellations like `TensorFiniteElement`